regulation-worker/internal/delete/batch/batch.go

```diff
@@ -144,115 +113,106 @@ func (*Batch) updateStatusTrackerFile(absStatusTrackerFileName, fileName string)
-		return "", fmt.Errorf("error while opening file, %w", err)
+		return "", nil, fmt.Errorf("error while downloading statusTrackerFile: %w", err)
 	}
-	defer tmpFilePtr.Close()
-	absPath, err := filepath.Abs(tmpFilePtr.Name())
+
+	f, err := os.OpenFile(trackerFilePath, os.O_RDWR, 0o644)
 	if err != nil {
-		pkgLogger.Errorf("error while getting absolute path: %v", err)
-		return "", fmt.Errorf("error while getting absolute path: %w", err)
+		return "", nil, fmt.Errorf("unable to open status tracker file")
 	}
-	err = b.FM.Download(context.TODO(), tmpFilePtr, completeFileName)
-	if err != nil {
-		if err == filemanager.ErrKeyNotFound {
-			pkgLogger.Debugf("file not found")
-			return absPath, nil
-		}
-		pkgLogger.Errorf("error while downloading object using file manager: %v", err)
-		return "", fmt.Errorf("error while downloading object using file manager: %w", err)
-	}
-	return absPath, nil
+
+	return f.Name(), f, nil
 }
 
-// decompresses .json.gzip files to .json & remove corresponding .json.gzip file
-func (b *Batch) decompress(compressedFileName string) (string, error) {
-	compressedFilePtr, err := os.OpenFile(compressedFileName, os.O_RDWR, 0o644)
-	if err != nil {
-		return "", fmt.Errorf("error while opening compressed file, %w", err)
-	}
-	defer compressedFilePtr.Close()
+func (b *Batch) cleanedFiles(_ context.Context, f io.ReadWriteCloser, job *model.Job) ([]string, error) {
+	pkgLogger.Debugf("fetching already cleaned files based on the status tracker file")
 
-	gzipReader, err := gzip.NewReader(compressedFilePtr)
-	if err != nil {
-		return "", fmt.Errorf("error while reading compressed file: %w", err)
-	}
+	var response []string
+	defer f.Close()
 
-	decompressedFilePtr, err := os.CreateTemp(b.TmpDirPath, "")
+	byt, err := io.ReadAll(f)
 	if err != nil {
-		return "", fmt.Errorf("error while creating temporary file for decompressed files during cleaning")
+		return nil, fmt.Errorf("unable to read contents of status tracker file, err: %w", err)
 	}
-	defer decompressedFilePtr.Close()
+	// if statusTracker.txt exists then read it & remove all those files name from above gzFilesObjects,
+	// since those files are already cleaned.
 
-	_, err = io.Copy(decompressedFilePtr, gzipReader)
-	if err != nil {
-		return "", fmt.Errorf("error while writing uncompressed file: %w", err)
-	}
+	jobID := fmt.Sprintf("%d", job.ID)
 
-	decompressedFileName, err := filepath.Abs(decompressedFilePtr.Name())
-	if err != nil {
-		return "", err
-	}
+	if len(byt) == 0 {
+		// insert <jobID> in 1st line
+		if _, err := io.WriteString(f, jobID+"\n"); err != nil {
+			return nil, fmt.Errorf("error while writing to jobId to statusTrackerFile: %w", err)
+		}
+	} else {
+		lines := strings.Split(string(byt), "\n")
+		// check if our <jobID> matches with the one in file.
+		// if not, then truncate the file & write new current jobID.
 
-	return decompressedFileName, nil
-}
+		// TODO: Why would this happen ? ( remanent of the past )
+		if lines[0] != jobID {
```

Comment:
As we discussed, this might be a problem if two jobs are assigned on the same bucket and:
1. job 1 starts, but gets interrupted (gets marked as pending and will be retried later)
2. then job 2 gets pickup 

We need two re-consider this case.

Reply:
I get it now. I'll make sure we have a ticket for properly reviewing this.